docs/internal-architecture.md

```diff
@@ -0,0 +1,53 @@
+## Internal architecture
+
+There are a few resources available to understand how the Collector works:
```

Comment:
I feel like this sentence does not add much. 

```suggestion
This document describes the Collector internal architecture and startup flow. It can be helpful if you are starting to contribute to the Collector codebase.

For the end-user focused architecture document, please see the [opentelemetry.io's Architecture documentation](https://opentelemetry.io/docs/collector/architecture/) 
```